opengever/maintenance/scripts/cleanup_todo_notifications.py

```diff
@@ -0,0 +1,82 @@
+"""
+This script deletes notifications for activities on ToDos that were incorrectly
+sent. These were due to a bug adding new participant of a workspace as Watcher
+on all ToDos and not just the ones in that particular workspace.
+
+Note that it is not possible to know which notifications were wrongfully sent,
+as we cannot know what users were legitimate watchers at the time of the
+Activity. Instead, we delete all notifications for ToDo activities that were
+sent to users who are not currently watchers on the corresponding resource.
+
+    bin/instance run ./scripts/cleanup_todo_notifications.py
+
+"""
+from opengever.activity.model import Activity
+from opengever.activity.model import Notification
+from opengever.base.model import create_session
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+import sys
+import transaction
+
+
+def find_notifications_to_delete():
+    to_delete = []
+    todo_activities = Activity.query.filter(
+        Activity.kind.in_(['todo-modified', 'todo-assigned'])
```

Comment:
this is not scoped by the current plone site. will that be an issue? Or is global cleanup desired?

Reply:
thanks. I did want to scope that... I'll change it (although the script is for TR deployments which have only one admin unit ;-))